### PR TITLE
Fix read_llc_to_tiles dimension order

### PR DESCRIPTION
**Pull request by timothyas**
Currently the dimension order of a given data set is not consistent across functions. With #42, `read_llc_to_tiles` reads in data and puts the tile dimension FIRST no matter what. That means 3D spatial data have dimensions [N_tiles, N_vertical, N_llc, N_llc], time varying 2D spatial data have dimensions [N_tiles, N_time, N_llc, N_llc], and time varying 3D spatial data have dimensions [N_tiles, N_time, N_vertical, N_llc, N_llc]. This is contradictory to `read_llc_to_tiles_xmitgcm`, which has the dimension order [N_time, N_vertical, N_tiles, N_llc, N_llc], and any singleton dimensions are dropped. I'll mention that this is why I previously had all of the swapaxes calls in this function - in order to be consistent with previous functionality and to be consistent with `llc_array_conversion` routines.

I would argue that the latter dimension order should be followed because (1) it seems more intuitive to me, (2) it follows the xmitgcm convention, which would be nice to agree so that a dataset read in through xmitgcm has the same underlying dimension order with what is read in with this routine, (3) less has to be changed with the current implementation of `llc_tiles_to_xda`.

This implements that change, and makes sure that singleton dimensions are squeezed out.

### CODE REVIEW

**Comment by ifenty** [comment]
Tim,

Let's agree to use the same convention as xmitgcm.  

3D = [time depth tile j i]
2D = [time tile j i] or [tile j i]  with tile as a coordinate in the dataset

I confirmed that this is what 'load_ecco_vars_from_mds' is doing now.  

I think this does make sense because compact files are stored as horizontal slices (with all 13 tiles), right?

Also, I noticed that xmitgcm convention seems to keep the singleton time dimension if one time record is loaded via (```open_mdsdataset```).  In  ```load_ecco_vars_from_mds``` now we squeeze at the end which removes the singleton time dimension.  I don't think it hurts anything to    keep the time dimension except that it does require the user to to squeeze or do ```X.isel(time=0``` before passing the `DataArray` to ecco.plot_tiles or similar routine which expects 13xllcxllc.

**Comment by timothyas** [comment]
Cool, I'm glad you agree on the convention. 

Yes you're right, `xmitgcm.open_mdsdataset` does leave singleton time dimensions. The squeeze was to make `read_llc_to_tiles` and `read_llc_to_tiles_xmitgcm` consistent. But after a little more testing it seems like the only time that they are inconsistent (when `read_llc_to_tiles_xmitgcm` squeezes a singleton dimension but `read_llc_to_tiles` does not) is while reading in a single vertical slice and multiple time records of a 4D field. Perhaps this case isn't really a concern, because then neither function really does what is desired anyway. 

TL; DR I got rid of the squeeze statement.
